vllm/model_executor/layers/attention.py

```diff
@@ -156,7 +156,8 @@ def single_query_cached_kv_attention(
         # sequences or heads is large, we use V1 since there is enough work
         # to parallelize.
         # TODO(woosuk): Tune this heuristic.
-        use_v1 = max_num_partitions == 1 or num_seqs * num_heads > 512
+        use_v1 = not input_metadata.force_paged_attention_v2 and (
+            max_num_partitions == 1 or num_seqs * num_heads > 512)
```

Comment:
Among the GPUs we support, T4 has the smallest shared memory, which is 48KB per SM. According to our calculation, this means the V1 kernel can support up to ~12K sequence length. So, why don't we simply enforce V2 when the max context len is greater than 8K?

This will be also good for the performance since we don't always use V2 even when the model's max len is large.

Reply:
Sounds good, let me do this!